src/unix/core.c

```diff
@@ -1153,3 +1153,113 @@ int uv_os_tmpdir(char* buffer, size_t* size) {
 
   return 0;
 }
+
+
+void uv_free_passwd(uv_passwd_t* pwd) {
+  if (pwd == NULL)
+    return;
+
+  uv__free(pwd->name);
+  uv__free(pwd->shell);
+  uv__free(pwd->homedir);
+}
+
+
+int uv_get_passwd(uv_passwd_t* pwd) {
+  struct passwd pw;
+  struct passwd* result;
+  char* buf;
+  uid_t uid;
+  size_t bufsize;
+  size_t len;
+  long initsize;
+  int r;
+#if defined(__ANDROID_API__) && __ANDROID_API__ < 21
+  int (*getpwuid_r)(uid_t, struct passwd*, char*, size_t, struct passwd**);
+
+  getpwuid_r = dlsym(RTLD_DEFAULT, "getpwuid_r");
+  if (getpwuid_r == NULL)
+    return -ENOSYS;
+#endif
```

Comment:
Can you move this code and the `uv_os_homedir()` code into a `uv__getpwuid_r()` function?